components/core/src/event.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::fs::{self, File};
+use std::io::{Write, Read};
+use std::path::Path;
+use std::time::{UNIX_EPOCH, SystemTime};
+use uuid::Uuid;
+use rustc_serialize::json::{ToJson, Json};
+use fs::cache_analytics_path;
+
+// Supported events
+pub const EVENT_BUILDER_PROJECT_CREATE: &'static str = "builder-project-create";
+
+// Sample event JSON payload (compatible with Segment.io)
+// {
+//   "type": "track",
+//   "event": "builder-project-create",
+//   "properties": {
+//     "clientid" : "0a5c0882-ade5-46cf-821d-8d3853cd0d41"
+//     "timestamp": "1479330000.13442404",
+//   }
+// }
+
+const CLIENT_ID_METAFILE: &'static str = "CLIENT_ID";
+
+#[derive(Debug, Clone)]
+pub struct Event {
+    name: String,
+    clientid: String,
+    timestamp: String,
+    properties: BTreeMap<String, String>,
```

Comment:
Are we structuring our Event struct like this because that is the format that Segment expects? We might want to format this however we see fit and then massage it to the event type that Segment expects with a `ToSegment` trait. That way we can fully leverage the type system in Rust.

If this is actually just a grab bag of extra data then ignore this comment ;)